frontend/src/pages/components/WebSocketCall.jsx

```diff
@@ -20,25 +20,34 @@ export default function WebSocketCall({ socket }) {
     if (!message) {
       return;
     }
-    socket.emit("data", {'username': sessionStorage.Username + ": ", 'message': message});
+    socket.emit("message", {'token': sessionStorage.token, 'content': message, 'public_id': sessionStorage.currentChat});
     setMessage("");
   };
 
+  socket.emit("join", {'token': sessionStorage.token})
   useEffect(() => {
-    socket.on("data", (data) => {
-      setMessages([...messages, data.data]);
+      socket.on("join", (data) => {
```

Comment:
oh ok yeah, all the list of chatrooms and all the messages to those chatrooms should be in this response. we should be saving the data received here into the state somehow so that we can use it.